harness/java/io/temporal/sdkfeatures/Main.java

```diff
@@ -1,29 +1,79 @@
 package io.temporal.sdkfeatures;
 
+import static picocli.CommandLine.*;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.google.common.base.Verify;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 import io.grpc.netty.shaded.io.netty.handler.ssl.SslContext;
 import io.micrometer.core.instrument.util.StringUtils;
 import io.temporal.serviceclient.SimpleSslContextBuilder;
-import picocli.CommandLine;
-
-import java.io.FileInputStream;
-import java.io.FileNotFoundException;
-import java.io.InputStream;
+import java.io.*;
+import java.net.Socket;
+import java.net.URI;
 import java.util.Arrays;
 import java.util.List;
 import java.util.NoSuchElementException;
-
-import static picocli.CommandLine.*;
-
 import javax.net.ssl.SSLException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine;
 
 @Command(name = "features", description = "Runs Java features")
 public class Main implements Runnable {
 
+  enum Outcome {
+    PASSED,
+    FAILED,
+    SKIPPED;
+  }
+
+  private static class SummaryEntry {
+    String name;
+    String outcome;
+    String message;
+
+    public SummaryEntry(String name, String outcome, String message) {
+      this.name = name;
+      this.outcome = outcome;
+      this.message = message;
+    }
+
+    public String getName() {
+      return name;
+    }
+
+    public String getOutcome() {
+      return outcome;
+    }
+
+    public String getMessage() {
+      return message;
+    }
+  }
+
+  BufferedWriter createSummaryServerWriter() {
```

Comment:
Ug, I still wish this was an HTTP server (metadata, status codes, content type, well supported, concurrent, req/resp if we want it, etc). But maybe we can tackle that at a later time.